src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1095,6 +1132,13 @@ Node *LShiftINode::Ideal(PhaseGVN *phase, bool can_reshape) {
       phase->type(add1->in(2)) == TypeInt::make( bits_mask ) )
     return new LShiftINode( add1->in(1), in(2) );
 
+  // Performs:
+  // (X << con1) << con2 ==> X << (con1 + con2) (see implementation for subtleties)
```

Comment:
```suggestion
  // (X << con1) << con2 ==> X << (con1 + con2)
```
I would keep it simple. It is usual that there are more comments in the implementation ;)